src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect } from 'react';
+import { useLazyQuery } from '@apollo/react-hooks';
+import { makeStyles } from '@material-ui/core/styles';
+import gql from 'graphql-tag';
+import Header from '../Header';
+import SearchBar from '../SearchBar';
+import AuthorResult from '../AuthorResult';
+
+const useStyles = makeStyles(() => ({
+  searchReply: {
+    margin: 'auto',
+    padding: '2rem',
+    color: 'coral',
+    fontWeight: 400,
+  },
+  divAfterHeader: {
+    height: '12vh',
+    minHeight: '12em',
+  },
+}));
+
+const SearchPage = () => {
+  const classes = useStyles();
+  const SEARCH_QUERY = gql`
+    query SearchAuthorQuery($author: String!) {
+      getPosts(page: 0, perPage: 25, filter: { author: $author }) {
+        title
+        published
+        url
+        feed {
+          id
+          author
+        }
+      }
+    }
+  `;
+
+  const [searchText, setSearchText] = useState('');
+  const [results, setResults] = useState([]);
+
+  const [executeSearch, { data }] = useLazyQuery(SEARCH_QUERY, {
+    // setting author variable here to use in the query above
+    variables: { author: searchText },
+
+    onCompleted: () => {
+      const returnedIds = [];
+      // We only want to display an author once, this would avoid the issue if
+      // there are multiple posts by the same author returned in our query
+      const finalResults = data.getPosts
+        .map(result => {
+          if (!returnedIds.includes(result.feed.id)) {
```

Comment:
This isn't right, `map()` needs to always return a value in its callback.  Sounds like you want to `filter()` first.